### PR TITLE
Potential fix for singularity eating itself

### PR DESCRIPTION
**Pull request by itsmeow**
## About The Pull Request

The singulo ate itself last round. I think this is how? Also adds some logging and sanity checks.

## Why It's Good For The Game

Fixes a bug.

## Testing Photographs and Procedure

The cause of this is really hard to identify, it has to move onto a turf it's queued to eat. Not sure how to test this.

## Changelog
:cl:
fix: Potentially fix singularities eating themselves.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Did it explode or just disappear? If it just disappears then its probably due to singularities losing mass over time.

File: code/datums/components/singularity.dm [hunk 1]
```diff
@@ -225,6 +225,8 @@
 				break
 			// eat the stuff if we're going to move into it so it doesn't mess up our movement
 			for(var/atom/thing_on_turf in current_turf.contents)
+				if(thing_on_turf == parent)
+					continue
 				consume(src, thing_on_turf)
 			consume(src, current_turf)
```
Comment (PowerfulBacon) [inline code/datums/components/singularity.dm:231]: This should be impossible unless something inside singularity_act sleeps, unless consuming a turf causes everything on it to be consumed too

**Comment by PowerfulBacon** [comment]
Logging for when a singularity eats another singularity is good though

**Comment by PowerfulBacon** [comment]
It most likely just ran out of energy rather than ate itself, especially if it was at the minimum level. I wasn't there to see it. I still think the changes are good for logging it, but the check inside moved() is pointless so long as its synchronous which it should be

**Comment by itsmeow** [comment]
> It most likely just ran out of energy rather than ate itself, especially if it was at the minimum level. I wasn't there to see it. I still think the changes are good for logging it, but the check inside moved() is pointless so long as its synchronous which it should be

Running out of energy is logged. I checked logs and there was no log. It broke containment, ate a containment field generator and a portion of the floor, and then boom it was gone.

File: code/modules/power/singularity/singularity.dm [hunk 2]
```diff
@@ -227,6 +227,9 @@
 	return 1
 
 /obj/anomaly/singularity/proc/consume(atom/thing)
+	if(thing == src)
+		stack_trace("consume() called on self by singularity [src]")
+		return
```
Comment (PowerfulBacon) [inline code/modules/power/singularity/singularity.dm]:
```suggestion
		CRASH("consume() called on self by singularity [src]")
```

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Crash is better if we want to return null since it won't result in the logs being combined into a single place

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**